snuba/clickhouse/formatter/expression.py

```diff
@@ -219,3 +220,31 @@ def _format_datetime_literal(self, exp: Literal) -> str:
 
     def _format_date_literal(self, exp: Literal) -> str:
         return "$D"
+
+    def _anonimize_alias(self, alias: str) -> str:
+        # there may be an alias that looks like `snuba_tags[something]`
+        # the string between the square brackets has the potential to be PII
+        # This function will anonimize that which is between the brackets.
+        # this may erroneously anonimize aliases with square brackets
+        # if they are input by the user, but that is better than leaking PII
+        between_square_brackets_regex = re.compile(r"(?<=\[).*?(?=\])")
```

Comment:
Compiling regex is expensive, please do this outside the function.

Reply:
Also why is this using lookaheads/behinds? Why not `\[(.+)\]`? Then you would capture the string to replace directly.

Reply:
> Also why is this using lookaheads/behinds? Why not \[(.+)\]? Then you would capture the string to replace directly.

two reasons:

1. The lookaheads/behinds returns just the string within the square brackets
2. This way you can support arbitrary indexing. E.g. `_snuba_tags[something][something_else]` I know we don't currently do that but it seemed the more future proof solution

Reply:
That makes sense. However regex already has support for what you're trying to do, by wrapping things with brackets to make capture groups.

```suggestion
        between_square_brackets_regex = re.compile(r"(?<=\[)(.*?)(?=\])") # this should still be outside the function
        anonymized_alias = regex.sub("$A", alias)
```

If the input is `_snuba_tags[something][something_else]`, then the output of this should be `_snuba_tags[$A][$A]`.